mozdownload/scraper.py

```diff
@@ -411,8 +416,10 @@ def get_build_info(self):
 
     def get_latest_build_date(self):
         """Return date of latest available nightly build."""
-        if self.application not in ('fennec'):
+        if self.application not in ('fenix', 'fennec'):
             url = urljoin(self.base_url, 'nightly', 'latest-%s/' % self.branch)
+        elif self.application == 'fenix':
+            url = urljoin(self.base_url, 'nightly/')
```

Comment:
Please note that this would cause a failure for Fenix because for this application we do not have a `latest` folder. Instead we would have to retrieve the folder for the he latest date by scraping the subfolders with the latest year and then month. Once found another parsing step is needed to retrieve the list of builds from which we then can extract the date.

It means that the code here needs to set the url to eg. `nightly/2023/01/` and then get the listing for the given platform and pick the latest entry for the date regex, eg. `2022-12-31-17-02-16-fenix-110.0b1-android-x86_64/`. This should extract `2022-12-31` as date. See similar code in `get_build_info()`. Then we also have to return early with the found date to not trigger an exception in line 433